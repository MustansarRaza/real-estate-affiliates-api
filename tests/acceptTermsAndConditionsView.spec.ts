import { loadSequelizeFixtures, useRequestFactory } from "./support";
import { ModelFixtureNames } from "./support/data";

describe("/acceptTermsAndConditions", () => {
  beforeEach(async () => {
    await loadSequelizeFixtures(
      ModelFixtureNames.AGENTS,
      ModelFixtureNames.USERS,
      ModelFixtureNames.CONTACTS,
      ModelFixtureNames.AFFILIATES,
      ModelFixtureNames.AFFILIATE_CONTACTS,
      ModelFixtureNames.POTENTIAL_AFFILIATE_VERIFICATION_STATUS,
      ModelFixtureNames.POTENTIAL_AFFILIATES
    );
  });

  it("updates user profile with accepted version of terms and conditions", async () => {
    const { body, header } = await useRequestFactory()
      .withVerifiedUser()
      .post("/acceptTermsAndConditions")
      .send({
        termsAndConditionsVersion: "1.1.1",
      })
      .expect(200);

    expect(body).toMatchSnapshot();
    expect(header).toMatchSnapshot();
  });

  it("returns 404 if no profile could be found for the user", async () => {
    const { body } = await useRequestFactory()
      .withNonExistantUser()
      .post("/acceptTermsAndConditions")
      .send({
        termsAndConditionsVersion: "1.1.1",
      })
      .expect(404);

    expect(body).toMatchSnapshot();
  });
});
