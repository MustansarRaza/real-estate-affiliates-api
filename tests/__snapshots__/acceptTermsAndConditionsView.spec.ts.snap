// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/acceptTermsAndConditions returns 404 if no profile could be found for the user 1`] = `
Object {
  "errors": Array [
    Object {
      "message": "Current user is not an affiliate",
      "name": "NotFound",
    },
  ],
}
`;

exports[`/acceptTermsAndConditions updates user profile with accepted version of terms and conditions 1`] = `
Object {
  "affiliate": Object {
    "agencyID": 1337,
    "assignees": Array [],
    "id": 1,
  },
  "affiliateVerificationStatus": Object {
    "id": 1,
    "status": "new",
  },
  "profile": Object {
    "address": "Disney Street 123",
    "affiliateType": "agency",
    "businessName": "Disney",
    "email": "walt@disney.com",
    "familyName": "Disney",
    "givenName": "Walt",
    "govIDNumber": "44443333",
    "govIDType": "NIC",
    "landLinePhoneNumbers": Array [],
    "locationID": 3,
    "mobilePhoneNumbers": Array [
      "+923344416658",
    ],
    "taxNumber": "128937934hsdhjdskf",
    "termsAndConditionsVersion": "1.1.1",
    "username": "walt@disney.com",
  },
  "ssoID": "4255c2ae-3f11-4a26-a401-3981c8845df3",
  "status": "verified",
}
`;

exports[`/acceptTermsAndConditions updates user profile with accepted version of terms and conditions 2`] = `
Object {
  "cache-control": "private, no-store",
  "connection": "close",
  "content-length": "553",
  "content-type": "application/json; charset=utf-8",
  "etag": "W/\\"229-vKpVlVZxQxrzw4RZzHck9vrHXKU\\"",
  "strict-transport-security": "max-age=15552000; includeSubDomains",
  "vary": "Accept-Encoding",
  "x-content-type-options": "nosniff",
  "x-dns-prefetch-control": "off",
  "x-download-options": "noopen",
  "x-frame-options": "SAMEORIGIN",
  "x-xss-protection": "1; mode=block",
}
`;
